galera/src/replicator_smm.cpp

```diff
@@ -2399,6 +2410,10 @@ void galera::ReplicatorSMM::process_vote(wsrep_seqno_t const seqno_g,
     local_monitor_.leave(lo);
 }
 
+wsrep_status_t galera::ReplicatorSMM::rotate_gcache_key() {
+    bool res = master_key_provider_.NotifyKeyRotationObserver();
```

Comment:
I think it doesn't matter because the observer is RingBuffer::RingBuffer. When it exists (and it exists always) it registers itself and returns true if encryption is disabled (rotation will cause an error) or false in case of encryption is enabled and rotation process success.
When RingBuffer is destroyed (destructor), the observer is set to the one returning `false` by default, and I think that part is wrong. In other words, for sanity, the default observer should return true (error) as it does not do any rotation. (but again: there is always RingBuffer observer registered, so only this part matters)

Reply:
changed default observer to return `true` (failure in rotation)